### PR TITLE
Add support of multiple batches per request

### PR DESCRIPTION
**Pull request by dmitryax**
otel-collector now can group multiple jaeger batches together and send them at once. This commit adds support for that.

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -109,11 +109,11 @@ func New(opts ...Option) (*Client, error) {
 	return c, nil
 }
 
-// Export takes a Jaeger batch and uses one of the available workers to export it synchronously.
+// Export takes a Jaeger batches and uses one of the available workers to export it synchronously.
 // It returns an error in case a request cannot be processed. It's up to the caller to retry.
-func (sa *Client) Export(ctx context.Context, batch *jaegerpb.Batch) error {
+func (sa *Client) Export(ctx context.Context, batches []*jaegerpb.Batch) error {
```
Comment (tigrannajaryan) [inline client/client.go:114]: This is a breaking change in a public function. Do we import this repo using a specific commit?
Perhaps to be safe just add an `ExportMultiple` function and keep the old one for now?

Comment (owais) [reply client/client.go:114]: The package is versioned and is fixed in the mod file so we can make a breaking change without breaking anyone's builds. Let's make sure to bump the major version to be clear.

File: client/worker.go [hunk 2]
```diff
@@ -52,16 +52,21 @@ func newWorker(client *http.Client, endpoint string, accessToken string) *worker
...
 
-	span.AddAttributes(trace.Int64Attribute("spans", int64(len(batch.Spans))))
-	if len(batch.Spans) == 0 {
+	var spansCount int
+	for _, batch := range batches {
+		spansCount += len(batch.Spans)
+	}
+
+	span.AddAttributes(trace.Int64Attribute("spans", int64(spansCount)))
+	if spansCount == 0 {
```
Comment (owais) [inline client/worker.go:65]: Can we add a `batches` count to the span as well?

File: client/worker.go [hunk 3]
```diff
@@ -52,16 +52,21 @@ func newWorker(client *http.Client, endpoint string, accessToken string) *worker
 	return w
 }
 
-func (w *worker) export(ctx context.Context, batch *jaegerpb.Batch) *ErrSend {
+func (w *worker) export(ctx context.Context, batches []*jaegerpb.Batch) *ErrSend {
 	_, span := trace.StartSpan(ctx, "export")
```
Comment (owais) [inline client/worker.go:56]: Unrelated but good oppurtunity to assign the returne context back to `ctx` instead of ignoring.

Comment (owais) [reply client/worker.go:56]:
```golang
 	ctx, span := trace.StartSpan(ctx, "export")
```

Comment (owais) [reply client/worker.go:56]: I see there are more places this needs to be fixed. If it is too many, I can open a PR on Monday to fix it. Feel free to ignore.

File: client/client.go (hunk 1 above)
Comment (dmitryax) [reply client/client.go:114]: Sure. we'll release it as 0.5

File: client/worker.go (hunk 2 above)
Comment (dmitryax) [reply client/worker.go:66]: added

File: client/worker.go (hunk 3 above)
Comment (dmitryax) [reply client/worker.go:56]: yes. looks like it deserves another PR

Comment (dmitryax) [reply client/worker.go]: actually I'm wrong, looks like it's the only please. changed

Comment (owais) [reply client/worker.go]: I think there should be a couple more in this file. At least one more in the send function.

Comment (dmitryax) [reply client/worker.go]: But the ctx not being used in those

Comment (owais) [reply client/worker.go]: yea but i think we should still not ignore it so that future code uses the correct context.

**Pull request merged**